File: leetcode/easy/check-if-string-is-a-prefix-of-array.py
# https://leetcode.com/problems/check-if-string-is-a-prefix-of-array/

class Solution:
    def isPrefixString(self, s: str, words: List[str]) -> bool:
        # MINE
        temp = ''
        for word in words:
            temp += word
            if temp == s:
                return True
            if not s.startswith(temp):
                return False
        return False

        # ANSWER

File: leetcode/easy/check-if-string-is-a-prefix-of-array.ts
// https://leetcode.com/problems/check-if-string-is-a-prefix-of-array/

// MINE
function isPrefixString(s: string, words: string[]): boolean {
  let temp = ''
  for (let word of words) {
    temp += word
    if (temp === s) return true
    if (!s.startsWith(temp)) return false
  }
  return false
}

// ANSWER
